extensions/arc/src/test/mocks/fakeContainersAndBuilders.ts

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as should from 'should';
+import { FakeRadioButton } from './fakeRadioButton';
+
+export const radioButtons: FakeRadioButton[] = [];
+
+export const divContainer = <azdata.DivContainer><any>{
+	withProperties(obj: azdata.DivContainerProperties) {
+		should(obj).not.be.undefined();
+		should(obj.clickable).not.be.undefined();
+		obj.clickable!.should.equal(false, 'withProperties need to be get called with clickable set to false');
+		return divContainer;
+	},
+	component: () => divContainer,
+	clearItems: () => { radioButtons.length = 0; },
+	addItem: (component: azdata.Component) => radioButtons.push(component as FakeRadioButton),
+	get items() { return radioButtons; }
+};
+
+export const loadingComponent = <azdata.LoadingComponent>{
+	get loading() { return false; },
+	set loading(_value: boolean) { }
+};
+
+const loadingBuilder = (<azdata.LoadingComponentBuilder><any>{
+	withItem(obj: any) {
+		obj.should.equal(divContainer);
```

Comment:
Yes, these are all object specific to the tests and not generic at all. Should I just move them to the corresponding test file?

Reply:
You might as well make them generic now - they'll be useful to use for other tests so no reason not to do that now so we don't have to come back and do that later on.

Reply:
I though it would be better to pay that cost when we need to use these for something else

Reply:
It's not really a cost at this point - it's what should have been done from the start and should be about the same amount of code. 

Part of the benefits of using TypeMoq is you can just fill in the parts you need and then others can easily add in other functionality later on as needed.

Reply:
The main benefit of TypeMoq is that it allows one to construct a mock object for an Interface which sinon does not allow. Otherwise, sinon is easier to work with and particularly easier to troubleshoot and debug when things do not work as per plan.

Reply:
Changed to using generic typemoq mock objects. Removed all sinon mocks which were specific to the tests being written. A nice result was that the test code has no typemoq/sinon code and the stub where all the typemoq code resides has on test knowledge.